src/lib/eZ/RichText/Resources/stylesheets/docbook/xhtml5/edit/core.xsl

```diff
@@ -585,7 +585,7 @@
       <xsl:call-template name="ezattribute"/>
       <xsl:apply-templates/>
       <xsl:if test="./docbook:ezattribute or not(node())">
-        <xsl:text> </xsl:text>
+        <xsl:text>ao-upcast-placeholder</xsl:text>
```

Comment:
This doesn't make much sense to me. What is `ao-upcast-placeholder` and why I would want to expose this in OE? Is this shown as attribute value?

Reply:
Sorry, I just choose this placeholder and made a type there. It should be `ae-upcast-placeholder`. Basically its value does not matter at all. It just needs to be some non-space characters. It is passed only to OE, and so AlloyEditor upcasts the widget for it.
Without this placeholder, there will be following HTML in the editor:
```
...
<p><span some-custom-tag-attributes /></p>
...
```
And AlloyEditor will ignore that custom tag span for widget upcasting. The point is to get some paceholder within the span:
```
...
<p><span some-custom-tag-attributes>ao-upcast-placeholder</span></p>
...
```
And in this case, the custom tag span will be upcasted and transformed to the widget. It's widget will not contain the placeholder text at all.

@alongosz should we change the placeholder text? Do you have any better suggestions for that?